packages/sdk/src/index.ts

```diff
@@ -229,3 +245,36 @@ export class Gomu {
     };
   }
 }
+
+async function approveAssetsForMarketplace({
+  marketplaceName,
+  marketplace,
+  assetOrOrder,
+}: {
+  marketplaceName: MarketplaceName;
+  marketplace: Marketplace<any>;
+  assetOrOrder: Asset | Order;
+}): Promise<AssetApprovalResponse> {
+  try {
+    const approvalPromise =
+      "marketplaceOrder" in assetOrOrder
+        ? marketplace.approveTakeOrderAsset(assetOrOrder)
+        : marketplace.approveAsset(assetOrOrder);
+
+    await approvalPromise;
+
+    return {
+      marketplaceName: marketplaceName as MarketplaceName,
+      data: {
+        approved: true,
+      },
+    };
+  } catch (err) {
+    return {
+      marketplaceName: marketplaceName as MarketplaceName,
+      error: {
+        message: err instanceof Error ? err.message : `${err}`,
```

Comment:
Can we create utils function that does this, since we have this line of code in 2 places already. Something like `retrieveErrorMessage()`.